docs/en/quality_en.md

```diff
@@ -0,0 +1,147 @@
+
+<p align="center">
+    <img src="../images/quality-theory.png" alt="quality-theory.png" width="650" />
+</p>
+
+  ### Procedures
+The quality of the code merged into the repository is controlled through strict procedures to ensure that each line of code merged will not affect the function of the main process before submitted to QA, and conform to the code style guide (Curve is developed using C++, and uses ``cpplint``  and ``cppcheck`` for code style checking).  We have already transfer the development of Curve to GitHub, and this procedure has been be transplanted.
+
+Two important steps should be mentioned: **manual reviewing and continuous integration (CI)**
+
+  <img src="../images/quality-control-1.png" alt="quality-control-1.png" width="850" />
+
+  <img src="../images/quality-control-2.png" alt="quality-control-2.png" width="850" />
+
+## Theories
+
+### Unit Test
+
+Unit test for testing the correctness of a module, a function, or a class, it will guarantee the correctness of module's behavior in a great extend, and serves as the most basic test in the procedure of software development. **Code coverage** is used to measure the completeness of the unit test.  Figure 4 below shows a statistical chart of Curve code coverage, calculated by the tool "lcov (for c++)", and rendered by "genhtml". Other languages such as go also have their own  Coverage statistics tool.  The unit tests of curve modules are under folder "curve/test".
+
+<p align="center">
+    <img src="../images/quality-cov.png" alt="quality-cov.png" width="950" /><br>
+    <font size=3>Figure 4 Curve code coverage result</font>
+</p>
+
+### Integration test
+
+#### necessity
+
+The issues to be considered in the integration test are mainly the problems after the modules are connected:
+
+- Whether the data passing through the module interface will be lost;
+- Whether the combination of sub-functions can meet the expectation;
+- Whether the sub-modules will affect each other;
+- Whether the error accumulation of a single module will be amplified to an unacceptable level; 
+
+Therefore, it is necessary to conduct a integration test to find and eliminate the problems above that may occur in the module connection after the unit test, and finally construct the required software subsystem or system.
+
+#### Content
+
+- **Function test**
+  
+  Function test is to conduct a comprehensive test on the functions provided by the module from user's prospective. Before the test, it is necessary to design sufficient test cases, considering various system states and parameter input to decide whether the module can still work as usual, and return the expected results. In this process, a scientific test cases design method is required to consider various input scenarios, and complete the test process with least cases and execution steps.
```

Comment:
whether the module can still work as usual,
->  whether the module can still work normally,